IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -110,7 +110,7 @@ + (BOOL)isValidIssuer:(NSString *)certAuths
 {
     NSString *regexString = @"OU=[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}";
     keychainCertIssuer = [keychainCertIssuer uppercaseString];
-    certAuths = [certAuths uppercaseString];
+    certAuths = [certAuths.msidURLDecode uppercaseString];
```

Comment:
Is msidURLDecode different from msidWWWFormURLDecode? I can see msidWWWFormURLDecode replacing + with a space first

Reply:
Good point. Thanks for bringing this up. However, they are different.
EX:
Input: 
OU%3d5ce5e2c6-f36e-4e53-840b-98acf367a25e%2cCN%3dMS-Organization-Access+%2cDC%3dMSIDLAB4+%2cDC%3dcom+%3bOU%3d82dbaca4-3e81-46ca-9c73-0950c1eaca97%2cCN%3dMS-Organization-Access+%2cDC%3dwindows+%2cDC%3dnet+

msidURLDecode output:
OU=5CE5E2C6-F36E-4E53-840B-98ACF367A25E,CN=MS-ORGANIZATION-ACCESS+,DC=MSIDLAB4+,DC=COM+;OU=82DBACA4-3E81-46CA-9C73-0950C1EACA97,CN=MS-ORGANIZATION-ACCESS+,DC=WINDOWS+,DC=NET+

msidWWWFormURLDecode:
OU=5CE5E2C6-F36E-4E53-840B-98ACF367A25E,CN=MS-ORGANIZATION-ACCESS ,DC=MSIDLAB4 ,DC=COM ;OU=82DBACA4-3E81-46CA-9C73-0950C1EACA97,CN=MS-ORGANIZATION-ACCESS ,DC=WINDOWS ,DC=NET

But yep. Technically, they both will work, since the subsequent regex matching will be able to grep the correct substr out of both cases.

I just decided to comply with how the latest MSAL dev branch handles both URL encode & decode, as they are more spec-compliant based on this PR - https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/350